include/userobjects/ClusteringUserObject.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "GeneralUserObject.h"
+
+class AuxiliarySystem;
+
+class ClusteringUserObject : public GeneralUserObject
+{
+
+public:
+  static InputParameters validParams();
+  ClusteringUserObject(const InputParameters & parameters);
+
+  virtual void execute() override;
+  virtual void initialize() override {};
+  virtual void finalize() override {};
+  int getExtraIntegerScore(libMesh::Elem * elem);
```

Comment:
```suggestion
  int getExtraIntegerScore(libMesh::Elem * elem) const;
```

Mark getters with `const` if they don't modify any stored data / do anything complicated.